src/amplitude-client.js

```diff
@@ -1552,4 +1598,33 @@ if (BUILD_COMPAT_2_0) {
  */
 AmplitudeClient.prototype.__VERSION__ = version;
 
+/**
+ * Determines whether or not to push call to this._q or invoke it
+ * @private
+ */
+AmplitudeClient.prototype._shouldDeferCall = function _shouldDeferCall() {
+  return this._pendingReadStorage || this._initializationDeferred;
+};
+
+/**
+ * Defers Initialization by putting all functions into storage until users
+ * have accepted terms for tracking
+ * @private
+ */
+AmplitudeClient.prototype._deferInitialization = function _deferInitialization(apiKey, opt_userId, opt_config, opt_callback) {
+  this._initializationDeferred = true;
+  this._q.push(['init', apiKey, opt_userId, opt_config, opt_callback]);
```

Comment:
we're never pushing init to `_q` and it's ultimately doing what we ne

Reply:
it doesn't seem like there's any potential risk to re-using it. I think it would add more confusion if we had multiple queues that are vaguely defined

Reply:
👍

Reply:
I hope this doesn't cause some weird circular queueing behavior where you defer a function that pushes to this or something, etc 🤔 (seems fine I guess?)

Reply:
i can't imagine it having circular dependency issues 🤔 
since it's a queue, it will run functions in order and it can tolerate if we pass in a function multiple times